### PR TITLE
Only use exported types in the plugin api

### PR DESCRIPTION
**Pull request by ssiloti**

### CODE REVIEW

File: include/libtorrent/extensions.hpp [hunk 1]
```diff
@@ -177,12 +176,12 @@ POSSIBILITY OF SUCH DAMAGE.
 #include "libtorrent/socket.hpp"
 #include "libtorrent/sha1_hash.hpp" // for sha1_hash
 #include "libtorrent/error_code.hpp"
+#include "libtorrent/session_handle.hpp"
+#include "libtorrent/torrent_handle.hpp"
```
Comment (arvidn) [inline include/libtorrent/extensions.hpp]: I think you could get away with forward declarations of these types, no?

File: include/libtorrent/extensions/lt_trackers.hpp [hunk 2]
```diff
@@ -35,10 +35,12 @@ POSSIBILITY OF SUCH DAMAGE.
 
 #ifndef TORRENT_DISABLE_EXTENSIONS
 
+#include "libtorrent/config.hpp"
+#include "libtorrent/torrent_handle.hpp"
```
Comment (arvidn) [inline include/libtorrent/extensions/lt_trackers.hpp]: same here

**Comment by arvidn** [comment]
looks good apart from the additional includes in the header files. I think you could get away with forward declarations in the header files and include torrent_handle.hpp in the .cpp files instead.
also, one build configuration failed to build (see the travis-ci output).

**Comment by arvidn** [comment]
I thought you could pass incomplete types by value in function declarations, as long as the typ is complete in the definition and call site. Is that not the case?

**Comment by ssiloti** [comment]
I just double checked and nope, value arguments cannot be incomplete types.

**Pull request merged**